tenacity/_asyncio.py

```diff
@@ -75,5 +74,12 @@ async def __anext__(self):
     def wraps(self, fn):
         fn = super().wraps(fn)
         # Ensure wrapper is recognized as a coroutine function.
-        fn._is_coroutine = asyncio.coroutines._is_coroutine
-        return fn
+
+        async def async_wrapped(*args, **kwargs):
+            return await fn(*args, **kwargs)
+
+        # Preserve attributes
+        async_wrapped.retry = fn.retry
+        async_wrapped.retry_with = fn.retry_with
```

Comment:
maybe add tests on this too just to be sure?

Reply:
I think this test is enough for ensuring the attributes:

https://github.com/jd/tenacity/blob/be9bccbb731d28ab0bcd8a4ee5acae70988bf593/tenacity/tests/test_asyncio.py#L94-L96

Still, perhaps it would be safer to have a stand-alone test for this.